### PR TITLE
S3UTILS-54 crrExistingObjects is compatible with both S3C and Artesca

### PR DESCRIPTION
**Pull request by nicolas2bert**
The Pull Request introduces these key changes:

- **Switch to S3 and Backbeat API**: crrExistingObjects now uses S3 and Backbeat APIs instead of direct metadata calls, improving security and maintainability but potentially slowing down performance due to added abstraction. It's expected to be minor, but will undergo testing.

- **Refactoring for testability**: The script has been refactored to make it more modular and easier to test, improving code clarity and allowing for better unit/functional testing.

- **Adding unit/functional tests with mocks**: Unit/functional tests using mocked S3 and Backbeat APIs have been added, providing a controlled environment for more comprehensive and reliable testing of the script's functionality.

Note: Additional modifications are required in both CloudServer and Backbeat:
https://scality.atlassian.net/browse/CLDSRV-501
https://scality.atlassian.net/browse/BB-483

**!! Warning !! : Breaking change for Zenko deployment only (XDM and Artesca)**: introduction of mandatory environment variables ACCESS_KEY, SECRET_KEY, and ENDPOINT

### CODE REVIEW

File: CRR/ReplicationStatusUpdater.js [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+    doWhilst, eachSeries, eachLimit, waterfall, series,
+} = require('async');
+const werelogs = require('werelogs');
+const { ObjectMD } = require('arsenal').models;
+
+const { setupClients } = require('./clients');
+
+const LOG_PROGRESS_INTERVAL_MS = 10000;
+const AWS_SDK_REQUEST_RETRIES = 100;
+const AWS_SDK_REQUEST_DELAY_MS = 30;
```
Comment (anurag4DSB) [inline CRR/ReplicationStatusUpdater.js]: Do we need these variables here? THey are defined in the other file and are being used there.

**Review by anurag4DSB — APPROVED** [review]
LGTM, thanks for the context on CS and BB.
As this is a breaking change for Artesca, and XDM, we should create a follow up ARTESCA ticket for the same(may be already handled/discussed)

File: CRR/ReplicationStatusUpdater.js (hunk 1 above)
Comment (nicolas2bert) [reply CRR/ReplicationStatusUpdater.js]: We don't, good catch.

File: tests/unit/CRR/crrExistingObjects.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        // Save the original process.env
+        originalEnv = { ...process.env };
+
+        // Reset the argv array to its default state
+        process.argv = ['node', 'yourscript.js'];
+        // Reset the mock to clear previous call history
+        process.exit.mockReset();
+
+        // Clear the mocks before each test
+        // ReplicationStatusUpdater.mockClear();
```
Comment (jonathan-gramain) [inline tests/unit/CRR/crrExistingObjects.js]: Is this needed?

**Review by jonathan-gramain — APPROVED** [review]

**Comment by nicolas2bert** [comment]
@bert-e approve

**Pull request merged**